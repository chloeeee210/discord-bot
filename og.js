const ogs = require('open-graph-scraper');
const { url } = require("../my-bot/commands/notion.js")
const { endPoint } = require('../config.json');
const notion = require('../my-bot/commands/notion.js');
// const options = { url: 'http://ogp.me/' };
function og(url) {
    console.log("잘 긁어옴!" + url)
}
// notion에서 가져온 url로 값 받아오기as

// og("https://twinstae.github.s/");
module.exports = {
    og

    // ogs(options)
    //     .then((data) => {
    //         const { error, result, response } = data;
    //         console.log('error:', error);  // This returns true or false. True if there was an error. The error itself is inside the results object.
    //         console.log('result:', result); // This contains all of the Open Graph results
    //         console.log('response:', response); // This contains the HTML of page
    //     })

    

}