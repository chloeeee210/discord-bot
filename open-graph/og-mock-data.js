const fireshipUrl = "https://www.youtube.com/watch?v=G1rOthIU-uo";
const fireshipOgResult = {
  ogSiteName: "YouTube",
  ogUrl: "https://www.youtube.com/watch?v=G1rOthIU-uo",
  ogTitle: "Redis in 100 Seconds",
  ogDescription:
    "Use the special link https://redis.info/fireship (or code: MATRIX200) to try Redis Enterprise Cloud to get a $200 credit, become part of a weekly raffle, and...",
  alIosAppStoreId: "544007664",
  alIosAppName: "YouTube",
  alIosUrl:
    "vnd.youtube://www.youtube.com/watch?v=G1rOthIU-uo&feature=applinks",
  alAndroidUrl:
    "vnd.youtube://www.youtube.com/watch?v=G1rOthIU-uo&feature=applinks",
  alWebUrl: "http://www.youtube.com/watch?v=G1rOthIU-uo&feature=applinks",
  ogType: "video.other",
  ogLocale: "ko-KR",
  ogDate: "2021-07-08",
  favicon: "https://www.youtube.com/s/desktop/b3b81e9c/img/favicon.ico",
  charset: "utf8",
  requestUrl: "https://www.youtube.com/watch?v=G1rOthIU-uo",
  success: true,
};
const inflearnUrl = "https://www.inflearn.com/course/스프링-입문-스프링부트";
const inflearnOgResult = {
  ogUrl:
    "https://www.inflearn.com/course/%EC%8A%A4%ED%94%84%EB%A7%81-%EC%9E%85%EB%AC%B8-%EC%8A%A4%ED%94%84%EB%A7%81%EB%B6%80%ED%8A%B8",
  ogType: "website",
  ogTitle:
    "[무료] 스프링 입문 - 코드로 배우는 스프링 부트, 웹 MVC, DB 접근 기술 - 인프런 | 강의",
  ogDescription:
    "스프링 입문자가 예제를 만들어가면서 스프링 웹 애플리케이션 개발 전반을 빠르게 학습할 수 있습니다.,  - 강의 소개 | 인프런...",
  ogSiteName: "인프런",
  ogLocale: "ko-KR",
  twitterCard: "summary",
  twitterTitle:
    "[무료] 스프링 입문 - 코드로 배우는 스프링 부트, 웹 MVC, DB 접근 기술 - 인프런 | 강의",
  twitterDescription:
    "스프링 입문자가 예제를 만들어가면서 스프링 웹 애플리케이션 개발 전반을 빠르게 학습할 수 있습니다.,  - 강의 소개 | 인프런...",
  twitterUrl:
    "https://www.inflearn.com/course/%EC%8A%A4%ED%94%84%EB%A7%81-%EC%9E%85%EB%AC%B8-%EC%8A%A4%ED%94%84%EB%A7%81%EB%B6%80%ED%8A%B8",
  twitterSite: "@inflearn",
  twitterCreator: "@inflearn",
  ogImage: {
    url: "https://cdn.inflearn.com/public/courses/325630/cover/56f635a3-3a44-4096-a16b-453ea1696b1a/325630-eng.png",
    width: null,
    height: null,
    type: "png",
  },
  twitterImage: {
    url: "https://cdn.inflearn.com/public/courses/325630/cover/56f635a3-3a44-4096-a16b-453ea1696b1a/325630-eng.png",
    width: null,
    height: null,
    alt: null,
  },
  favicon: "/favicon.ico",
  charset: "utf8",
  requestUrl: "https://www.inflearn.com/course/스프링-입문-스프링부트",
  success: true,
};

export { fireshipUrl, fireshipOgResult, inflearnUrl, inflearnOgResult }