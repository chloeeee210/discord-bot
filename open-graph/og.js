import ogs from 'open-graph-scraper';
// const { Options } = require('discord.js');
// const options = { url: 'http://ogp.me/' };


export const ogScraper = (url) => {
    const options = {
      url,
      downloadLimit: 100000000
    };
    
    return ogs(options)
      .then((data) => {
        const { error, result, response } = data;
        // console.log('error:', error);  // This returns true or false. True if there was an error. The error itself is inside the results object.
        // console.log('result:', result); // This contains all of the Open Graph results
        return result;
        // console.log('response:', response); // This contains the HTML of page
      }).catch( e => {
        throw Error('open-graph-error : ' + e.message )
      })//에러를 받는 함수
}
