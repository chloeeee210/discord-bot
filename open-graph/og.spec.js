import { describe, it, expect } from "vitest";
//import tagger from "./og-tagging.js";
import { ogScraper } from "./og.js";
import {fireshipUrl, fireshipOgResult, inflearnUrl, inflearnOgResult} from './og-mock-data.js'
// 아니 이것도 분리해야할것같은데? ogScraper가 가장 상위고 그다음 tagging result에 대한 테스트가
// 아 !!!!!!!!!!!!!! 과거의 나!!!!!!!!!! 

describe("ogScraper", () => {
  it("if request to inflearn, then return inflearn og result", async () => {
    const result = await ogScraper(inflearnUrl);
    expect(result).toMatchSnapshot();
  });

  it("if request to youtube, then return youtue og result", async () => {
    const result = await ogScraper(fireshipUrl);
    expect(result).toMatchSnapshot();
  });
});
