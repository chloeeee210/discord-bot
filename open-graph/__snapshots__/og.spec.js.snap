// Vitest Snapshot v1

exports[`ogScraper > if request to inflearn, then return inflearn og result 1`] = `
{
  "charset": "utf8",
  "favicon": "/favicon.ico",
  "ogDescription": "스프링 입문자가 예제를 만들어가면서 스프링 웹 애플리케이션 개발 전반을 빠르게 학습할 수 있습니다.,  - 강의 소개 | 인프런...",
  "ogImage": {
    "height": null,
    "type": "png",
    "url": "https://cdn.inflearn.com/public/courses/325630/cover/56f635a3-3a44-4096-a16b-453ea1696b1a/325630-eng.png",
    "width": null,
  },
  "ogLocale": "ko-KR",
  "ogSiteName": "인프런",
  "ogTitle": "[무료] 스프링 입문 - 코드로 배우는 스프링 부트, 웹 MVC, DB 접근 기술 - 인프런 | 강의",
  "ogType": "website",
  "ogUrl": "https://www.inflearn.com/course/%EC%8A%A4%ED%94%84%EB%A7%81-%EC%9E%85%EB%AC%B8-%EC%8A%A4%ED%94%84%EB%A7%81%EB%B6%80%ED%8A%B8",
  "requestUrl": "https://www.inflearn.com/course/스프링-입문-스프링부트",
  "success": true,
  "twitterCard": "summary",
  "twitterCreator": "@inflearn",
  "twitterDescription": "스프링 입문자가 예제를 만들어가면서 스프링 웹 애플리케이션 개발 전반을 빠르게 학습할 수 있습니다.,  - 강의 소개 | 인프런...",
  "twitterImage": {
    "alt": null,
    "height": null,
    "url": "https://cdn.inflearn.com/public/courses/325630/cover/56f635a3-3a44-4096-a16b-453ea1696b1a/325630-eng.png",
    "width": null,
  },
  "twitterSite": "@inflearn",
  "twitterTitle": "[무료] 스프링 입문 - 코드로 배우는 스프링 부트, 웹 MVC, DB 접근 기술 - 인프런 | 강의",
  "twitterUrl": "https://www.inflearn.com/course/%EC%8A%A4%ED%94%84%EB%A7%81-%EC%9E%85%EB%AC%B8-%EC%8A%A4%ED%94%84%EB%A7%81%EB%B6%80%ED%8A%B8",
}
`;

exports[`ogScraper > if request to youtube, then return youtue og result 1`] = `
{
  "alAndroidAppName": "YouTube",
  "alAndroidPackage": "com.google.android.youtube",
  "alAndroidUrl": "vnd.youtube://www.youtube.com/watch?v=G1rOthIU-uo&feature=applinks",
  "alIosAppName": "YouTube",
  "alIosAppStoreId": "544007664",
  "alIosUrl": "vnd.youtube://www.youtube.com/watch?v=G1rOthIU-uo&feature=applinks",
  "alWebUrl": "http://www.youtube.com/watch?v=G1rOthIU-uo&feature=applinks",
  "charset": "utf8",
  "favicon": "https://www.youtube.com/s/desktop/dd76c683/img/favicon.ico",
  "ogDate": "2021-07-08",
  "ogDescription": "Use the special link https://redis.info/fireship (or code: MATRIX200) to try Redis Enterprise Cloud to get a $200 credit, become part of a weekly raffle, and...",
  "ogImage": {
    "height": "720",
    "type": "jpg",
    "url": "https://i.ytimg.com/vi/G1rOthIU-uo/maxresdefault.jpg",
    "width": "1280",
  },
  "ogLocale": "ko-KR",
  "ogSiteName": "YouTube",
  "ogTitle": "Redis in 100 Seconds",
  "ogType": "video.other",
  "ogUrl": "https://www.youtube.com/watch?v=G1rOthIU-uo",
  "ogVideo": {
    "height": "720",
    "type": "text/html",
    "url": "https://www.youtube.com/embed/G1rOthIU-uo",
    "width": "1280",
  },
  "requestUrl": "https://www.youtube.com/watch?v=G1rOthIU-uo",
  "success": true,
  "twitterAppIdGooglePlay": "com.google.android.youtube",
  "twitterAppIdiPad": "544007664",
  "twitterAppIdiPhone": "544007664",
  "twitterAppNameGooglePlay": "YouTube",
  "twitterAppNameiPad": "YouTube",
  "twitterAppNameiPhone": "YouTube",
  "twitterAppUrlGooglePlay": "https://www.youtube.com/watch?v=G1rOthIU-uo",
  "twitterAppUrliPad": "vnd.youtube://www.youtube.com/watch?v=G1rOthIU-uo&feature=applinks",
  "twitterAppUrliPhone": "vnd.youtube://www.youtube.com/watch?v=G1rOthIU-uo&feature=applinks",
  "twitterCard": "player",
  "twitterDescription": "Use the special link https://redis.info/fireship (or code: MATRIX200) to try Redis Enterprise Cloud to get a $200 credit, become part of a weekly raffle, and...",
  "twitterImage": {
    "alt": null,
    "height": null,
    "url": "https://i.ytimg.com/vi/G1rOthIU-uo/maxresdefault.jpg",
    "width": null,
  },
  "twitterPlayer": {
    "height": "720",
    "stream": null,
    "url": "https://www.youtube.com/embed/G1rOthIU-uo",
    "width": "1280",
  },
  "twitterSite": "@youtube",
  "twitterTitle": "Redis in 100 Seconds",
  "twitterUrl": "https://www.youtube.com/watch?v=G1rOthIU-uo",
}
`;
